### PR TITLE
making the kaldi import more robust

### PR DESCRIPTION
**Pull request by vesis84**
get_duration():
- recover if audio file cannot be loaded for get_duration(), drop such recordings...
- use chunksize for ProcessPoolExecutor::map (avoid hanging of ProcessPoolExecutor for large RecordingSets)

### CODE REVIEW

**Comment by vesis84** [comment]
Hi Piotr, what would you think about this change ?
Let's talk...

(I found this issue while importing per-utterance flac files for Chime challenge)
Cheers,
Karel

File: lhotse/kaldi.py [hunk 1]
```diff
@@ -122,10 +133,30 @@ def fix_id(t: str) -> str:
...
+        with ProcessPoolExecutor(max_workers=num_jobs) as ex:
+            dur_vals = list(
+                ex.map(get_duration, recordings.values(), chunksize=chunksize)
+            )
+
         durations = dict(zip(recordings.keys(), dur_vals))
 
+    # remove recordings with 'None' duration (i.e. there was a read error)
+    for recording_id, duration in durations.items():
+        if durations == None:
```
Comment (pzelasko) [inline lhotse/kaldi.py:151]:
```suggestion
        if duration is None:
```

**Review by pzelasko — COMMENTED** [review]
Looks good to me, I think I found one bug though (see the other comment) -- can you test it?

File: lhotse/kaldi.py [hunk 2]
```diff
@@ -122,10 +133,30 @@ def fix_id(t: str) -> str:
...
+
         durations = dict(zip(recordings.keys(), dur_vals))
 
+    # remove recordings with 'None' duration (i.e. there was a read error)
+    for recording_id, dur_value in durations.items():
+        if dur_value == None:
+            logging.warning(
+                f"[{recording_id}] Could not get duration. "
+                f"Failed to read audio from `{recordings[recording_id]}`. "
+                f"Dropping the recording from manifest."
```
Comment (csukuangfj) [inline lhotse/kaldi.py:155]:
```suggestion
                "Dropping the recording from manifest."
```

File: lhotse/kaldi.py [hunk 3]
```diff
@@ -122,10 +133,30 @@ def fix_id(t: str) -> str:
...
+        with ProcessPoolExecutor(max_workers=num_jobs) as ex:
+            dur_vals = list(
+                ex.map(get_duration, recordings.values(), chunksize=chunksize)
+            )
+
         durations = dict(zip(recordings.keys(), dur_vals))
 
+    # remove recordings with 'None' duration (i.e. there was a read error)
+    for recording_id, dur_value in durations.items():
+        if dur_value == None:
```
Comment (csukuangfj) [inline lhotse/kaldi.py:151]: To fix the style issue, we can use
```
        if dur_value is None:
```
to replace
```
        if dur_value == None:
```

**Comment by vesis84** [comment]
Ok, both suggested changes are done. I also added a new sanity check...
Cheers,
K.

**Pull request merged**